python/graphstorm/graphpartitioning/metadata_schema.py

```diff
@@ -0,0 +1,799 @@
+        if self._instance is not None:
+            return
+
+        self._basedir = None
+        self._metadata_path = None
+        self._data = None
+        self._global_nid_offsets = None
+        self._global_eid_offsets = None
+        self._ntypes = None
+        self._ntype_id_map = None
+        self._id_ntype_map = None
+        self._etype_id_map = None
+        self._id_etype_map = None
+        self._etypes = None
+        self._ntype_features = None
+        self._etype_features = None
+        self._ntype_feature_files = None
+        self._etype_feature_files = None
+        self._etype_files = None
+
+    def _load_json(self, file_name, base_dir):
+        """Helpper function to read the metadata schema file of the input
+        graph dataset. ``file_name`` input argument can be either in absolute
+        or relative path formats. ``base_dir`` is used as the root directory
+        for the input graph dataset where all the files (of the dataset) are
+        expected to be located on the disk.
+
+        Argument
+        --------
+        file_name: string
+            Filename of the input metadata schema file. This can be either
+            in absolute or relative path formats. If in relative format,
+            ``base_dir`` is used as the root directory.
+        base_dir: string
+            Base directory, if provided, is used as the root directory for
+            the input graph dataset
+        """
+        if file_name is None or not isinstance(file_name, str):
+            raise ValueError(f"File path is not valid. = {file_name}")
+
+        self._basedir = None
+        if base_dir is not None and os.path.isdir(base_dir):
+            if not os.path.isfile(os.path.join(base_dir, file_name)):
+                raise ValueError(
+                    f"Metadata file does not exist. {base_dir}, {file_name}"
+                )
+            self._metadata_path = os.path.join(
+                os.path.join(base_dir, file_name)
+            )
+            self._basedir = base_dir
+        else:
+            if not os.path.isfile(file_name):
+                raise ValueError(f"Metadata file does not exist. {file_name}")
+            self._metadata_path = file_name
+
+        if self._basedir is None:
+            self._basedir = os.getcwd()
+
+        with open(self._metadata_path, "r", encoding="utf-8") as handle:
+            self._data = json.load(handle)
+
+    def _check_ifexists(self, data_files, src_type):
+        """Helper function to verify the presence of a set of filenames.
+
+        Argument
+        --------
+        data_files: list of str
+            A set of file names, either in absolute or in relative formats.
+        src_type: str
+            Either node type or edge type from the input graph.
+        """
+        if not isinstance(data_files, list):
+            raise ValueError(
+                f"Incorrect value encountered when iterating over files for {src_type}."
+            )
```

Comment:
Again, if `data_files` is not a list, Python will raise the appropriate error.

Reply:
Done.